packages/maskbook/src/extension/background-script/EthereumServices/request.ts

```diff
@@ -1,13 +1,41 @@
+import { noop } from 'lodash-es'
 import type { RequestArguments } from 'web3-core'
 import type { JsonRpcPayload, JsonRpcResponse } from 'web3-core-helpers'
-import { INTERNAL_send, SendOverrides } from './send'
+import { INTERNAL_nativeSend, INTERNAL_send, SendOverrides } from './send'
+import { hasNativeAPI, nativeAPI } from '../../../utils/native-rpc'
+import { EthereumMethodType, ProviderType } from '@masknet/web3-shared'
+import { currentProviderSettings } from '../../../plugins/Wallet/settings'
+import { Flags } from '../../../utils'
+import { WalletRPC } from '../../../plugins/Wallet/messages'
+import { openPopupsWindow } from '../HelperService'
 
 let id = 0
 
+const unconfirmedCallbackMap = new Map<number, (error: Error | null, response?: JsonRpcResponse) => void>()
+
+function getSendMethod() {
+    if (hasNativeAPI && nativeAPI) return INTERNAL_nativeSend
+    return INTERNAL_send
+}
+
+function getPayloadId(payload: JsonRpcPayload) {
+    return typeof payload.id === 'string' ? Number.parseInt(payload.id, 10) : payload.id
+}
+
+function isRpcNeedToBeConfirmed(method: EthereumMethodType) {
```

Comment:
suggest, `rpc` all uppercase

```suggestion
function isRPCNeedToBeConfirmed(method: EthereumMethodType) {
```
```suggestion
function isRpcNeedToBeConfirmed(method: EthereumMethodType) {
```